packages/runtime/src/runtime/ensembleMenu.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useCallback, useState } from "react";
+import { Drawer as AntDrawer } from "antd";
+import {
+  type EnsembleAction,
+  type Expression,
+  type EnsembleWidget,
+  useRegisterBindings,
+} from "@ensembleui/react-framework";
+// eslint-disable-next-line import/no-cycle
+import { useEnsembleAction } from "./hooks";
+import { EnsembleRuntime } from "./runtime";
+
+type MenuTypes = "Drawer"; // More menu types can be added here and should be supported in the future
+
+export interface MenuModel {
+  type: MenuTypes;
+  id?: string;
+  onCollapse?: EnsembleAction;
+  title?: Expression<string>;
+  children?: EnsembleWidget[];
+  header?: EnsembleWidget;
+  footer?: EnsembleWidget;
+  styles?: { [key: string]: unknown };
+}
+
+export interface DrawerViewModel {
+  id?: string;
+  onCollapse?: EnsembleAction;
+  title?: Expression<string>;
+  children?: EnsembleWidget[];
+  header?: EnsembleWidget;
+  footer?: EnsembleWidget;
+  styles?: { [key: string]: unknown };
+}
+
+type Position = "left" | "right" | "top" | "bottom";
+const positions = ["left", "right", "top", "bottom"] as const;
+
+export const EnsembleMenu: React.FC<MenuModel> = (props) => {
```

Comment:
There should only be one component responsible for rendering the menu(s), either this one or `menu.tsx`.